group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2131,7 +2471,29 @@ public void replay(
      *
      * @return The result that contains records to append if the join group phase completes.
      */
-    public CoordinatorResult<Void, Record> classicGroupJoin(
+    public CoordinatorResult<Void, Record> groupJoin(
```

Comment:
I feel like that it would be great if we could keep the previous name here because the classic protocol is used in all cases.